woocommerce/woocommerce-android
# IAP: make it possible to optionally provide sandbox url for testing

Closes: #7927


Do not merge till https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2542 is merged and fluxc hash's changed. You can test it though via local build

### Description

* Adds the possibility to use sandbox base URL for mobile pay APIs
* Initial idea to use fluxc inside the IAP module doesn't work well, because it'll require making it possible to work with local builds as well, that's why providing the network communication now on the app

### Testing instructions

* Build debug build. Make a purchase and notice that network call is not done, but stubbed backed is used
* Build release build. Make a purchase and notice that the network call is not done, and real backed was hit
* add `iap_testing_sandbox_url = https://{your running sandbox}/sandboxed-api`  to your gradle.properies. Make purchase and notice that sandbox API is hit
